Core_Pregnancy/PregnancyGameController.cs

```diff
@@ -0,0 +1,303 @@
+                    controller.ReadData();
+            }
+
+        #elif AI
+
+            private static void ApplyToAllDatas(Func<AgentData, PregnancyData, bool> action)
+            {
+                var heroineList = GetHeroineList();
+                if (heroineList == null) return;
+
+                foreach (var heroine in heroineList)
+                {
+                    foreach (var chaFile in heroine.GetRelatedChaFiles())
+                    {
+                        var data = ExtendedSave.GetExtendedDataById(chaFile, PregnancyPlugin.GUID);
+                        var pd = PregnancyData.Load(data) ?? new PregnancyData();
+                        if (action(heroine, pd))
+                            ExtendedSave.SetExtendedDataById(chaFile, PregnancyPlugin.GUID, pd.Save());
+                    }
+                }
+
+                // If controller exists then update its state so it gets any pregnancy week updates
+                foreach (var controller in FindObjectsOfType<PregnancyCharaController>())
+                    controller.ReadData();
+            }
+
+
+            private static List<AgentData> GetHeroineList()
+            {
+                var heroineList = new List<AgentData>();
+                var agentTable = Manager.Map.Instance.AgentTable;            
+                if (agentTable == null || agentTable.Count <= 0) return null;
+
+                foreach (int key in agentTable.Keys)
+                {
+                    if (agentTable.TryGetValue(key, out AgentActor agentActor))
+                    {
+                        heroineList.Add(agentActor.AgentData);
+                    }
+                } 
+
+                return heroineList;
+            }
+        #endif
+
+
+
+        private static bool AddPregnancyWeek(PregnancyData pd)
+        {
+            if (pd == null || !pd.GameplayEnabled) return false;
+
+            if (pd.IsPregnant)
+            {
+                #if KK
+                    if (pd.Week < PregnancyData.LeaveSchoolWeek)
+                    {
+                        // Advance through in-school at full configured speed
+                        var weekChange = PregnancyPlugin.PregnancyProgressionSpeed.Value;
+                        pd.Week = Mathf.Min(PregnancyData.LeaveSchoolWeek, pd.Week + weekChange);
+                    }
+                    else if (pd.Week < PregnancyData.ReturnToSchoolWeek)
+                    {
+                        // Make sure at least one week is spent out of school
+                        var weekChange = Mathf.Min(PregnancyData.ReturnToSchoolWeek - PregnancyData.LeaveSchoolWeek - 1, PregnancyPlugin.PregnancyProgressionSpeed.Value);
+                        pd.Week = pd.Week + weekChange;
+                    }
+
+                    if (pd.Week >= PregnancyData.ReturnToSchoolWeek)
+                        pd.Week = 0;
+
+                #elif AI
+
+                    //We probaly can't make characters "leave school" in AI
+
```

Comment:
I don't think the "Leave School" mechanic applies well to AI so I've removed it.  It could have interfered with custom character scenes